src/tests/net.rs

```diff
@@ -6,68 +6,92 @@ use rand::{
 };
 
 use crate::{
-    alkahest, read_packet, write_packet_to_vec, Formula, Lazy, SerIter, Serialize, SerializeRef,
+    alkahest, read_packet, write_packet_to_vec, Deserialize, Formula, Lazy, SerIter, Serialize, SerializeRef, Ref,
 };
 
+#[alkahest(Formula)]
+pub enum GameMessageFormula {
+    Client(Ref<ClientMessageFormula>),
+    Server(Ref<ServerMessageFormula>),
+}
+
 #[derive(Debug, Clone, PartialEq, Eq)]
-#[alkahest(Formula, Serialize, Deserialize)]
+#[alkahest(Serialize<GameMessageFormula>, Deserialize<'_, GameMessageFormula>)]
 pub enum GameMessage {
     Client(ClientMessage),
     Server(ServerMessage),
 }
 
 #[derive(Debug)]
-#[alkahest(Deserialize<'de, GameMessage>)]
+#[alkahest(Deserialize<'de, GameMessageFormula>)]
 pub enum GameMessageRead<'de> {
     Client(ClientMessageRead<'de>),
     Server(ServerMessageRead<'de>),
 }
 
+#[alkahest(Formula)]
+pub enum ClientMessageFormula {
+    ClientData { nickname: Ref<str>, clan: Ref<str> },
```

Comment:
`Ref<str>` and `String` are distinct but actually equivalent formulas.

Reply:
Right :D Which is cool

Reply:
So why making separate type? `ClientMessage` as `Formula` would be the same.

Reply:
It was not obvious to me from documentation what a formula actually *is* -- that it is a rust type that mirrors the type you want that contains types which have a direct exact representation. Making this split it becomes more obvious what's going on -- in my eyes at least. And it shows how you can create multiple formulas which the same type can serialize as / deserialize from, or customize the exact representation with wrappers like `Ref`. The point of all this change was to try what it would look like to 1. separate all the formulas 2. make tweaks like adding `Ref` to build my intuition on the versioned wrapper we talked about in the other thread.

Whether the proper place for that is in a test like here: perhaps not x) but I was just playing around with it to get a better idea of how things work.

Probably it should go in a separate test/example to show it better. I would be fine with reverting the formula changes in this file for sure.